src/plugins/data_source_management/public/components/data_source_selector/data_source_selector.tsx

```diff
@@ -100,15 +117,17 @@ export class DataSourceSelector extends React.Component<
         ? 'Select a data source'
         : this.props.placeholderText;
 
-    const dataSources = this.props.dataSourceFilter
-      ? this.state.allDataSources.filter((ds) => this.props.dataSourceFilter!(ds))
-      : this.state.allDataSources;
+    const dataSources = getFilterDataSources(
```

Comment:
instead of having 2 separate functions and have one of them called twice, would it be better to have one function which takes the filter function and does the filtering logic and store the default data source in the state here, so that we just need to call the function once here?